spec/system/admin/orders_spec.rb

```diff
@@ -583,56 +583,49 @@ def extract_pdf_content
-                expect(page).to have_content "Please wait until the PDF is ready " \
-                                             "before closing this modal."
+              expect(page).to have_content "Compiling Invoices"
+              expect(page).to have_content "Please wait until the PDF is ready " \
+                                           "before closing this modal."
 
-                # we don't run Sidekiq in test environment, so we need to manually run enqueued jobs
-                # to generate PDF files, and change the modal accordingly
-                perform_enqueued_jobs(only: BulkInvoiceJob)
+              # we don't run Sidekiq in test environment, so we need to manually run enqueued jobs
+              # to generate PDF files, and change the modal accordingly
+              perform_enqueued_jobs(only: BulkInvoiceJob)
 
-                expect(page).to have_content "Bulk Invoice created"
+              expect(page).to have_content "Bulk Invoice created"
 
-                within ".modal-content" do
-                  expect(page).to have_link(class: "button", text: "VIEW FILE", href: /invoices/)
+              within ".modal-content" do
+                expect(page).to have_link(class: "button", text: "VIEW FILE", href: /invoices/)
 
-                  invoice_content = extract_pdf_content
+                invoice_content = extract_pdf_content
 
-                  expect(invoice_content).to have_content("TAX INVOICE", count: 2)
-                  expect(invoice_content).to have_content(order4.number.to_s)
-                  expect(invoice_content).to have_content(order5.number.to_s)
-                  expect(invoice_content).to have_content(distributor4.name.to_s)
-                  expect(invoice_content).to have_content(distributor5.name.to_s)
-                  expect(invoice_content).to have_content(order_cycle4.name.to_s)
-                  expect(invoice_content).to have_content(order_cycle5.name.to_s)
-                end
+                expect(invoice_content).to have_content("TAX INVOICE", count: 2)
+                expect(invoice_content).to have_content(order4.number.to_s)
+                expect(invoice_content).to have_content(order5.number.to_s)
+                expect(invoice_content).to have_content(distributor4.name.to_s)
+                expect(invoice_content).to have_content(distributor5.name.to_s)
+                expect(invoice_content).to have_content(order_cycle4.name.to_s)
+                expect(invoice_content).to have_content(order_cycle5.name.to_s)
               end
             end
           end
 
-          it_behaves_like "can bulk print invoices from 2 orders", "enabled"
-          it_behaves_like "can bulk print invoices from 2 orders", "disabled"
+          it_behaves_like "can bulk print invoices from 2 orders"
+
+          context "with legal invoices feature", feature: :invoices do
+            it_behaves_like "can bulk print invoices from 2 orders"
+          end
```

Comment:
Thanks @mkllnk, indeed this is the best option. I did not think of this here, although I believe we've used it in our tests before... anyway, thanks! :pray: